src/validation_workflow/docker/inspect_docker_image.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import json
+import logging
+import subprocess
+from subprocess import PIPE
+
+import requests
+
+from validation_workflow.validation_args import ValidationArgs
+
+"""
+This class is to compare the image digest/SHA-256 on OS and OSD between opensearchproject(downloaded/local) and opensearchstaging(on dockerHub).
+It retuns False if the digest does not match.
+"""
+
+
+class InspectDockerImage:
+
+    def __init__(self, image_id: str, image_name: str, prod_image_tag: str) -> None:
+        self.image_id = image_id
+        self.image_name = image_name
+        self.prod_image_tag = prod_image_tag
+        self.image_tag = ValidationArgs().stg_tag('OSD').replace(" ", "") if ("dashboards" in self.image_name) else ValidationArgs().stg_tag('OS').replace(" ", "")
+        self.auth_token_url = "https://auth.docker.io/token?"
+        self.auth_service_scope = "service=registry.docker.io&scope=repository:"
+        self.registry_url = "https://index.docker.io/v2/"
+        self.access_type = 'Bearer'
+
+    def inspect_digest(self) -> bool:
+
+        logging.info('Fetching access token from dockerHub')
+        api_url = self.auth_token_url + self.auth_service_scope + self.image_name.replace("opensearchproject", "opensearchstaging") + ":pull"
+        response = requests.get(api_url)
+        response_dict = json.loads(response.text)
+
+        logging.info('Fetching mainfest from DockerHub')
+        api_url = self.registry_url + self.image_name.replace("opensearchproject", "opensearchstaging") + "/manifests/" + self.image_tag
+        access_token = response_dict['token']
+
+        # set up all necessary token ( if you use VPN, it may slow down the response from dockerHub )
+        headersToken = {
+            "Authorization": f'{self.access_type} {access_token}',
+            "Accept": "application/vnd.docker.distribution.manifest.v2+json",
+            "Content-Type": "application/json;charset=UTF-8"
+        }
+
```

Comment:
Where is the token coming from?

Reply:
I mean to say which user? Are you hardcoding the token or you have a way to allow users enter that, or you can retrieve after user logging in?

Reply:
Thanks for the asking, @peterzhuamazon , at L39-41, we retrieve the token at everytime the script is run ( if I am not mistaken, the token expired in 1 hour); and the L31-32 defines the token_url to be used in API call.